src/cluster_legacy.c

```diff
@@ -5585,9 +5585,13 @@ void clusterBeforeSleep(void) {
             if (!(server.cluster_module_flags & CLUSTER_MODULE_FLAG_NO_FAILOVER)) clusterHandleReplicaFailover();
         }
     } else if (flags & CLUSTER_TODO_HANDLE_FAILOVER) {
-        /* Handle failover, this is needed when it is likely that there is already
+        /* Handle failover as soon as possible so that won't have a 100ms
+         * as it was handled only in clusterCron. This is needed when it
+         * is likely that we can start the election or there is already
          * the quorum from primaries in order to react fast. */
-        clusterHandleReplicaFailover();
+        if (nodeIsReplica(myself)) {
+            if (!(server.cluster_module_flags & CLUSTER_MODULE_FLAG_NO_FAILOVER)) clusterHandleReplicaFailover();
```

Comment:
Why nested if? One if is enough.

```suggestion
        if (nodeIsReplica(myself) &&
            !(server.cluster_module_flags & CLUSTER_MODULE_FLAG_NO_FAILOVER)) {
            clusterHandleReplicaFailover();
```

Reply:
yean, that is right. I just copied the above content (you can expand it to see it).

I previously wrote a simpler IF block that combined HANDLE_MANUALFAILOVER and HANDLE_FAILOVER, but later changed it to the current one.

Reply:
Yes I see it can be simplified, but we don't have to change too much. As you want.